### PR TITLE
Verifies "down" area behavior

### LINKED ISSUE #21: Check "behavior:down" in the area breakpoint algorithm
## What
Conduct a check and add tests to ensure `behavior:down` areas are working properly.

## Roadmap
- [ ] Investigate
- [x] (Optional) Fix
- [x] Add tests

### PR DESCRIPTION
**Pull request by kettanaito**
> First of all, thank you for your contribution! Please fill in this pull request template to let others know more about the awesome changes you have made. You rock!

## Issues

> (Optional) Please list the issues related or covered by this pull request. Otherwise remove this section.

* Fixes #21 

### Release version

> Please choose which release version should include your changes.

- [ ] major (backward-*in*compatible changes)
- [ ] minor (backward-compatible changes)
- [x] patch (non-breaking feature or bugfix)

### Changelog

> Please list the current or upcoming changes introduced by this pull request. In case of intermediate pull request, mark the finished changes so the roadmap is evident. 

- [x] Improves unit tests for `getAreasList` util
- [x] Uses object spreading for components object composition in `getComponents`

### CODE REVIEW

File: src/utils/getComponents/getComponents.js [hunk 1]
```diff
@@ -65,8 +65,9 @@ export default function getComponents({ areas, templates }: TAreasList) {
       ? AreaComponent
       : withPlaceholder(AreaComponent, areaParams)
 
-    return Object.assign({}, components, {
+    return {
+      ...components,
```
Comment (MatthiasMargot) [inline src/utils/getComponents/getComponents.js]: ES6 for the win 👍

Comment (kettanaito) [reply src/utils/getComponents/getComponents.js]: Turned out we are already using rest-spread support through Babel. No need for explicit `Object.assign` :)

**Comment by kettanaito** [comment]
This is an intermediate pull request. That means that it may cover the requested functionality just partially. I would need to have a look at it once more, since it's been some time from the last session.

New unit tests is a good thing.

**Comment by kettanaito** [comment]
Just found an unexpected behavior during breakpoint (area params) concatenation.

It's related to the analyzis of a breakpoint on which the area is not present:

```js
<Composition
  template="a"
  templateMd="b"
  templateLg="a" />
```

Now what the algorithm must do here, it should understand that `a` area is not present in `templateMd`, but there is a whole `sm` breakpoint in-between, where `a` *must* be still rendered. Thus, it should modify the previous `a` area param entry and stretch it to the beginning of `md` breakpoint, not including the breakpoint itself. This step is referred to as "area stretching".

The key point here, is that previously it stretched the area *and* replaced the last area with it. This is completely wrong. Any template definition where the area is not included explicitly must return `null` to the area params list. I will merge the correct logic today, as a part of this pull request.

**Comment by kettanaito** [comment]
@MatthiasMargot I think this pull request is ready to be merged from my side. Please, if you would have time, could you review it? If not, just tell me. Thanks.

**Comment by kettanaito** [comment]
@MatthiasMargot Thanks for the review.

Yes, I'm working on #28, which implies re-writing of `getAreaParams` method. Can't say it's less cluttered now, but it will have more clear steps, which can be separated into pure functions.

**Pull request merged**